### PR TITLE
feat: [REV-2847] add ecommerce/redeem_enrollment_code incoming connection & signals

### PR DESCRIPTION
**Pull request by pshiu**
## Description

We are doing an internal hackathon for some rapid prototyping in the Coordinator. The objective of this hackathon is to get proof of concept code for redeeming enrollment codes (aka 100% coupons) via an existing ecommerce system in our company called Titan.

This PR adds:
* An incoming external connection through a new Coordinator API resource at `/ecommerce/redeem_enrollment_code`
* A internal connection through a new signal called `enrollment_code_redemption_requested_signal` and its receiver `enrollment_code_redemption_requested_create_order`
* A Celery task called `enrollment_code_redemption_requested_create_order_task` that will in the future call the Titan `/redeemEnrollmentCode` endpoint in a future PR in this repo.

## Additional information

* Jira: [REV-2847](https://2u-internal.atlassian.net/browse/REV-2847)
* Previous art: https://github.com/edx/commerce-coordinator/pull/44
* Draft sequence diagram:
    ```mermaid
    sequenceDiagram
        actor user as User
        participant lms as edX LMS
        participant ecommerce as edX Ecommerce
        participant frontend as Enrollment Page Frontend
        participant coordinator as edX Coordinator
        participant titan as Titan
        user->>frontend: GET {enrollment code URL}
        frontend->>+coordinator: POST /redeem_enrollment_code
        coordinator->>-titan: POST /redeemEnrollmentCode
        titan->>+coordinator: GET /checkEnrollmentCode
        coordinator->>+ecommerce: GET /coupons/check/[id]
        ecommerce-->>-coordinator: /coupons/check/[id] response
        coordinator-->>-titan: /checkEnrollmentCode response
        titan->>+coordinator: POST /fulfill
        coordinator->>ecommerce: POST /coupons/invalidate/[id]
        coordinator->>-lms: POST /api/enrollment/v1/enrollment
    ```
* Objective of this PR is to implement `frontend->>+coordinator: POST /redeem_enrollment_code` and add the wiring for `coordinator->>-titan: POST /redeemEnrollmentCode` to happen in the Coordinator.

## Testing instructions

It is a known issue this PR has no tests. Tests will be created after the proof of concept hackathon.

### CODE REVIEW

File: commerce_coordinator/apps/ecommerce/signals.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"""
+Ecommerce signals and receivers.
+"""
+import logging
+
+from commerce_coordinator.apps.core.signal_helpers import CoordinatorSignal
+
+logger = logging.getLogger(__name__)
```
Comment (julianajlk) [inline commerce_coordinator/apps/ecommerce/signals.py:8]: Logger is here for future use?

File: commerce_coordinator/apps/ecommerce/urls.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"""
+ecommerce app URLS
+"""
+
+from django.urls import include, path
+
+from commerce_coordinator.apps.ecommerce.views import RedeemEnrollmentCodeView
+
+app_name = 'ecommerce'
```
Comment (julianajlk) [inline commerce_coordinator/apps/ecommerce/urls.py:9]: Should this match the name in `apps.py` (aka `commerce_coordinator.apps.ecommerce`)

File: commerce_coordinator/apps/ecommerce/views.py [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
+from rest_framework.response import Response
+from rest_framework.throttling import UserRateThrottle
+from rest_framework.views import APIView
 
-# Create your views here.
+from commerce_coordinator.apps.core.signal_helpers import format_signal_results
+
+from .signals import enrollment_code_redemption_requested_signal
+
+logger = logging.getLogger(__name__)
```
Comment (julianajlk) [inline commerce_coordinator/apps/ecommerce/views.py:15]: Same as above, not being used?

File: commerce_coordinator/apps/ecommerce/urls.py (hunk 2 above)
Comment (julianajlk) [reply commerce_coordinator/apps/ecommerce/urls.py:9]: I see in `urls.py` that there's `ecommerce' and under that, do we have `ecommerce`, `frontend_app_ecommerce` and `titan`?

File: commerce_coordinator/apps/ecommerce/signals.py (hunk 1 above)
Comment (pshiu) [reply commerce_coordinator/apps/ecommerce/signals.py:8]: Yep!

File: commerce_coordinator/apps/ecommerce/views.py (hunk 3 above)
Comment (pshiu) [reply commerce_coordinator/apps/ecommerce/views.py:15]: Yep!

**Review by julianajlk — APPROVED** [review]
LGTM! Just some minor comments on loggers not being used

File: commerce_coordinator/apps/ecommerce/signals.py (hunk 1 above)
Comment (julianajlk) [reply commerce_coordinator/apps/ecommerce/signals.py:8]: Is that necessary? Wouldn't it be better to introduce them when we actually need them

Comment (pshiu) [reply commerce_coordinator/apps/ecommerce/signals.py:8]: We can. I will remove them in a subsequent PR!

**Pull request merged**